pkg/devspace/config/loader/patch/operation.go

```diff
@@ -35,38 +34,66 @@ func (op *Operation) Perform(doc *yaml.Node) error {
 		return err
 	}
 
-	if len(matches) == 0 {
-		if op.Op == opAdd {
-			matches, err = getParents(doc, op.Path)
-			if err != nil {
-				return fmt.Errorf("could not add using path: %s", op.Path)
+	if len(matches) == 0 && op.Op != opAdd {
+		return fmt.Errorf("%s operation does not apply: doc is missing path: %s", op.Op, op.Path)
+	}
+
+	var f func(parent *yaml.Node, match *yaml.Node)
```

Comment:
Please rename this to a more meaningful variable name and make sure to add a comment why this is needed